umbraco/Umbraco-CMS
# Add option to keep the mini profiler active at all times in debug mode

### Prerequisites

- [x] I have [created an issue](https://github.com/umbraco/Umbraco-CMS/issues) for the proposed changes in this PR, the link is: https://github.com/umbraco/Umbraco-CMS/issues/3458

### Description

This PR opens up for adding `umbracoProfilerAlwaysActive` to the app settings. If that app setting is `true`, the mini profiler will be active for all frontend requests no matter if `umbDebug` is in the query string or not.

Obviously this only works in debug mode 😉 

The app setting is optional.

To test this:

1. Add `<add key="umbracoProfilerAlwaysActive" value="true" />` to the app settings in web.config.
2. Verify that all frontend requests are profiled.
3. Change the value of the app setting to `false`.
4. Verify that frontend requests are only profiled when using `umbDebug` in the query string.
5. Remove `umbracoProfilerAlwaysActive` from the app settings.
6. Verify that the site doesn't break, and that frontend requests can still be profiled using `umbDebug` in the query string.

If this PR is accepted I'll make sure to update the [web.config docs](https://our.umbraco.com/documentation/Reference/Config/webconfig/) and the [debugging docs](https://our.umbraco.com/Documentation/Getting-Started/Code/Debugging/) on Our as well.
